components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/StorageMaintenanceWorkerTest.kt

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.sync
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.work.Data
+import androidx.work.ListenableWorker
+import androidx.work.await
+import androidx.work.testing.TestListenableWorkerBuilder
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import mozilla.components.support.test.rule.runTestOnMain
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+import kotlin.reflect.KVisibility
+
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class StorageMaintenanceWorkerTest {
+
+    @get:Rule
+    val mainCoroutineRule = MainCoroutineRule()
+
```

Comment:
We should clear our provider so every test case is starting with a clean slate.

```suggestion
    @After
    fun cleanup() {
        GlobalPlacesDependencyProvider.placesStorage = null
    }
```